http/src/lib.rs

```diff
@@ -581,12 +575,50 @@ impl Server {
 			executor.wait();
 		}
 	}
+
+	/// Returns a handle to the server that can be used to close it while another thread is
+	/// blocking in `wait`.
+	pub fn close_handle(&mut self) -> CloseHandle {
```

Comment:
The function shouldn't mutate anything, so it should be `&self`.

```suggestion
	pub fn close_handle(&self) -> CloseHandle {
	  let executor_close: Option<Vec<_>> = self.executor.as_ref().map(|executors| {
			executors
				.iter()
				.map(|executor| executor.close_handle())
				.collect()
		});
		...
	}
```